include/envoy/http/codec.h

```diff
@@ -85,8 +85,8 @@ class StreamEncoder {
 class RequestEncoder : public virtual StreamEncoder {
 public:
   /**
-   * Encode headers, optionally indicating end of stream. Response headers must
-   * have a valid :status set.
+   * Encode headers, optionally indicating end of stream. HTTP/1.1 request headers must have a host
```

Comment:
This interface is for both HTTP/1.x and HTTP/2.
Do you think it makes sense to add the specific comment only to the HTTP/1.x method (maybe with an `ASSERT(host != nullptr)` as this is an invariant)?

Reply:
I think there is an assertion on status still?
Also can you remind me where the host assertion comes from?   We could weigh removing it - HTTP/0.9 doesn't require a host for example and we have to serialize raw HTTP to integration test it.

Reply:
Hmm yeah, I'll need to remove that comment (see @alyssawilk review). I think this is only relevant to the CONNECT without host header issue, and I'll need to know if CONNECT without host is allowed (if it is I'll remove the comment and I think the resolution here is to fix this line https://github.com/envoyproxy/envoy/blob/bad35c86655c77c155303c1c4b9f651f7de42714/source/common/http/http1/codec_impl.cc#L380